src/main/java/net/fabricmc/loom/configuration/ifaceinject/InterfaceInjectionProcessor.java

```diff
@@ -210,10 +224,19 @@ public static List<InjectedInterface> fromMod(FabricModJson fabricModJson) {
 			final List<InjectedInterface> result = new ArrayList<>();
 
 			for (String className : addedIfaces.keySet()) {
-				final JsonArray ifaceNames = addedIfaces.getAsJsonArray(className);
+				final JsonArray ifacesInfo = addedIfaces.getAsJsonArray(className);
+
+				for (JsonElement ifaceElement : ifacesInfo) {
+					String ifaceInfo = ifaceElement.getAsString();
+
+					if (ifaceInfo.contains("<") && ifaceInfo.contains(">")) {
+						String ifaceName = ifaceInfo.substring(0, ifaceInfo.indexOf("<"));
+						String generics = ifaceInfo.substring(ifaceInfo.indexOf("<"));
```

Comment:
Can we do some basic validation of the generics here? ASM has a `CheckSignatureAdapter` that looks somewhat useful.

It may be even possible to validate some additional stuff when applying to the class, such as checking that `T` is present on the target class? Im not too worried about this.